aws/resource_aws_api_gateway_rest_api_test.go

```diff
@@ -203,6 +203,24 @@ func TestAccAWSAPIGatewayRestApi_EndpointConfiguration(t *testing.T) {
 			},
 		},
 	})
+
+	resource.Test(t, resource.TestCase{
```

Comment:
We generally prefer to create whole new top level acceptance test functions instead of adding multiple `resource.Test()` within the same function -- this allows us to parallelize our testing better. 😄 

Another complicated piece to our API Gateway testing here is that we run the service testing across multiple AWS partitions (Commercial and GovCloud (US)), which the latter may or may not support the new `PRIVATE` REST API endpoint type. If you look at the `TestStep` above your addition here, we have a `PreConfig` function that ensures the AWS API Gateway service its calling supports the `EDGE` endpoint type or skips the rest of the test. We'll want to add something similar for `PRIVATE` as well.

Do not be discouraged here though! There just happens to be more advanced testing handling with this resource testing than other parts of our resource testing. You actually were on the right track except the tweaks mentioned above, which I can handle in a commit after yours to expedite releasing this enhancement for everyone. 👍

Reply:
Ok, I get it. I just didn't know private type may not be supported in some regions.

As for the remaing, I understand you're going to make the adjustments yourself, right? Anyway, feel free to ask for it, if you want to. I think I could handle it. ;)